share/lib/hoc/import3d/read_swc.hoc

```diff
@@ -292,36 +356,83 @@ proc mksections() {local i, j, isec, first  localobj sec, psec, pts
 	mksection(isec, first, i)
 }
 
-proc mksection() { local i, isec, first   localobj sec
-	isec = $1  first=$2  i=$3
-	if (isec > 0) {// branches have pid as first point
-		sec = new Import3d_Section(first, i-first+1)
-		pt2sec(pix2ix(first), sec.parentsec)
-		// but if the parent is the root and the branch has more than
-		// one point, then connect to center of root with wire
-		if (point2sec.x[pix2ix(first)] == 0 && i > 1) {
-			sec.parentx = 0.5
-			sec.first = 1
-		}else{
-			if (pix2ix(first) == 0) { sec.parentx = 0 }
-		}
-		sec.append(0, pix2ix(first), 1, x, y, z, d)
-		sec.append(1, first, i-first, x, y, z, d)
-	}else{// pid not first point in root section
-		sec = new Import3d_Section(first, i-first)
-		sec.append(0, first, i-first, x, y, z, d)
-	}
-	sec.type = type.x[first]
-	sections.append(sec)
-	if (object_id(sec.parentsec)) {
-		if (sec.parentsec.type == 1 && sec.type != 1) {
-			sec.d.x[0] = sec.d.x[1]
-		}
-	}
-	if (connect2prox.x[first]) {
-		sec.pid = sec.parentsec.id
-		sec.parentx = 0
-	}
+proc mksection() { local i, isec, first, den_con_soma, con_soma, handled   localobj sec, psec
+  isec = $1  first=$2  i=$3
+  if (isec == 0) {// Root. pid not first point in root section
+    if (soma3geom) { // treat as single point sphere
+      i = 1
+    }
+    sec = new Import3d_Section(first, i-first)
+    sec.append(0, first, i-first, x, y, z, d)
+  }else{ // Not root. Branches have pid as first point
+    sec = new Import3d_Section(first, i-first+1)
+    pt2sec(pix2ix(first), sec.parentsec)
+    psec = sec.parentsec
+
+    // 1) If 1 point soma, all branches connect with wire to parentx=0.5;
+    // 2) All branches that connect to last point of parent have parentx=1;
+    // 3) All branches that connect to first point of root, then parentx=0;
+
+    // 4) If branch is not soma and connects to interior of type 1 root, use
+    //    pt3dtype of 1 and parentx=0.5 (best if parent point is middle point)
+    //    with x,y,z of the parent point.
+
+    // 5) If branch is not soma and parent is soma and a child soma is
+    //    connected to that point also, use a pt3dtype of 1 and
+    //    parentx = 0 or 1 depending on whether first or last point
+    //    and x,y,z of the parent point.
+
+    den_con_soma = ((psec.type == 1) && (type.x[first] != 1))
+    con_soma = (psec.type == 1)
+    handled = 0
+    if (psec == sections.o(0)) { // connect to root
+      handled = 1
+      if (den_con_soma && psec.d.size == 1) { // parent is single point soma
+        sec.parentx = 0.5
+        if (i - first > 1) { // connect by wire
+          sec.first = 1
+        }
+      } else if (pix2ix(first) == psec.id) { // connect to first point of root
+        sec.parentx = 0.0
+        if (type.x[first] != 1 && nchild_soma.x(pix2ix(first)) > 1) {
```

Comment:
Am I correct to think that there's no difference between `vec.x[i]` and `vec.x(i)`?

Nitpick: I think stylistically it would be better to pick one (probably square brackets) and be consistent. Line 397 currently uses both syntaxes.

Reply:
No difference in this context but you are correct stylistically and I changed to square brackets.